src/vtt-segment-loader.js

```diff
@@ -976,6 +980,22 @@ export default class VTTSegmentLoader extends videojs.EventTarget {
 
     segment.requested = true;
 
+    // prepend the media initialization segment if it exists
+    if (segment.map) {
+      const initId = initSegmentId(segment.map);
+      const initSegment = this.initSegments_[initId];
+
+      segment.map.bytes = initSegment.bytes;
+
+      const vttHeader = new Uint8Array('WEBVTT\n\n'.split('').map(char => char.charCodeAt(0)));
+      const combinedByteLength = vttHeader.byteLength + segmentInfo.bytes.byteLength;
+      const combinedSegment = new Uint8Array(combinedByteLength);
+
+      combinedSegment.set(vttHeader);
+      combinedSegment.set(segmentInfo.bytes, vttHeader.byteLength);
+      segmentInfo.bytes = combinedSegment;
```

Comment:
I'd probably say (3). One, since it is a parser-specific requirement, we can do it right near where we pass it into the parser (it also makes it easier to change if the parser requirements ever loosen/we switch parsers). Would also be good to have a comment explaining it there. In addition, I'd rather us append to the init segment because I think it is more likely in the wild that we're going to have to look at some weird segment bytes/parsing errors, and if we don't modify them it'll make it easier for us to look into them and understand the parsing error.